### PR TITLE
Make != and NOT_in filters public

### PR DESCRIPTION
**Pull request by thebrianchen**
In addition to making these public, I also added NOT_IN to the `isInequality()` check. This means that queries using NOT_IN will be ordered by the field name, as is the case with inequality filters. Updated tests to reflect the change and made changed the inequality filter validation message to match the other SDKs.

### CODE REVIEW

File: firebase-firestore/src/androidTest/java/com/google/firebase/firestore/QueryTest.java [hunk 1]
```diff
@@ -640,19 +638,21 @@ public void testQueriesCanUseNotInFilters() {
...
+    expectedDocsMap.remove("f");
+    expectedDocsMap.remove("i");
+    expectedDocsMap.remove("j");
 
     QuerySnapshot snapshot =
         waitFor(collection.whereNotIn("zip", asList(98101L, 98103L, asList(98101L, 98102L))).get());
     assertEquals(Lists.newArrayList(expectedDocsMap.values()), querySnapshotToValues(snapshot));
 
-    // With objects.
+    //  With objects.
```
Comment (wilhuff) [inline firebase-firestore/src/androidTest/java/com/google/firebase/firestore/QueryTest.java:651]: nit: extra whitespace

**Review by wilhuff — APPROVED** [review]
LGTM

File: firebase-firestore/src/androidTest/java/com/google/firebase/firestore/QueryTest.java (hunk 1 above)
Comment (thebrianchen) [reply firebase-firestore/src/androidTest/java/com/google/firebase/firestore/QueryTest.java]: removed.

**Pull request merged**